### PR TITLE
document new Python API

### PR DESCRIPTION
**Pull request by zeke**
This PR updates the Cog Python API docs.

It's piggybacking on PR https://github.com/replicate/cog/pull/378

### CODE REVIEW

**Comment by zeke** [comment]
cc @bfirsh @andreasjansson

File: docs/python.md [hunk 1]
```diff
@@ -22,23 +36,25 @@ class Predictor(BasePredictor):
...
+
+This _required_ method is where you call the model that was loaded during `setup()`, but you may also want to add pre- and post-processing code here.
 
-The `predict()` function takes an arbitrary list of named arguments, where each argument name must correspond to a `@cog.input()` annotation.
+The `predict()` method takes an arbitrary list of named arguments, where each argument name must correspond to an [`Input()`](#inputkwargs) annotation.
 
-`predict()` can output strings, numbers, `pathlib.Path` objects, or lists or dicts of those types. We are working on support for other types of output, but for now we recommend using base-64 encoded strings or `pathlib.Path`s for more complex outputs.
+`predict()` can return strings, numbers, `pathlib.Path` objects, or lists or dicts of those types. You can also define a custom [`Output()`](#outputbasemodel) for more complex return types.
 
 #### Returning `pathlib.Path` objects
```
Comment (bfirsh) [inline docs/python.md:59]: We aren't returning `pathlib.Path` objects any longer -- we're returning `cog.Path` (or `cog.File`) objects.

A better heading for this section would perhaps be "Returning files" or "Outputting files".

File: docs/python.md [hunk 2]
```diff
@@ -47,22 +63,58 @@ If the output is a `pathlib.Path` object, that will be returned by the built-in
...
+- `cog.Path`: a path to a file on disk
+
+## `Output(BaseModel)`
+
+Your `predict()` method can return a simple data type like a string or a number, or a more complex object with multiple values.
+
+You can optionally use cog's `Output()` function to define the object returned by your `predict()` method:
+
+```py
+from cog import BasePredictor, BaseModel
```
Comment (bfirsh) [inline docs/python.md:111]: This wasn't in there so I added it :D https://github.com/replicate/cog/pull/378/commits/7e989203e39ce2c21f5731187f7472894e5ffa1e

**Review by bfirsh — APPROVED** [review]
Great! Broadly looks good and we can fix up later.

File: docs/python.md (hunk 2 above)
Comment (zeke) [reply docs/python.md:111]: Nice! I actually kinda knew that and hoped this document would will it into existence. :trollface:

**Comment by zeke** [comment]
@bfirsh I was about to make some followup edits to this doc based on your feedback but don't see these changes on your fork's branch.

I see "bfirsh merged commit 626d2e9 into bfirsh:new-cog-4-real 3 hours ago" above, but looking at that branch, the commit is nowhere in sight: https://github.com/bfirsh/cog/commits/new-cog-4-real

Am I missing something? 🤔

**Comment by bfirsh** [comment]
_blink_ not sure what hapepned there. Probably some force pushing gone awry. It's back now...

**Pull request merged**